### DISCUSSION TITLE
`UnitReference.get` method return type mismatch

### DISCUSSION THREAD

**Discussion by mbrunoviegas** [automock/automock#439] [Q&A] [answered]
Hi! I'm facing a weird behavior using NestJs and it's related `@suites` packages.

I reproduced the problem in a simple application, which you can check [here](https://github.com/mbrunoviegas/cqrs-nestjs).

I have the following test file:

```ts
import type { Mocked } from '@suites/doubles.jest';
import { TestBed } from '@suites/unit';
import { BasicQueryHandler } from './basic-query.handler';
import { AppService } from './app.service';
import { BasicQuery } from './queries/basic-query';
import { APP_SERVICE, IAppService } from './interfaces/app.service';

describe('BasicQueryHandler', () => {
  let handler: BasicQueryHandler;
  let appService: Mocked<AppService>;

  beforeEach(async () => {
    const { unit, unitRef } =
      await TestBed.solitary(BasicQueryHandler).compile();
    handler = unit;
    appService = unitRef.get<IAppService>(APP_SERVICE);
  });

  it('should return "Hello World!"', async () => {
    appService.getHello.mockReturnValue('Hello MB!');
    const result = await handler.execute(new BasicQuery('World'));
    expect(result).toBe('Hello MB!');
  });
});
```

The first execution is sucessful and this is the doc generated from the library about `unitRef.get`

![NoError](https://github.com/user-attachments/assets/b9f9f1bb-4711-4326-a8cc-d4bba142b4ae)

After installing `@faker-js/faker`, an error was reported, indicating that the return typing of `unitRef.get` was different from the one declared for `appService`. And also the description of `get` method, changed.

```
Type 'StubbedInstance<IAppService>' is not assignable to type 'Mocked<AppService>'.
  Type 'StubbedInstance<IAppService>' is not assignable to type '{ getHello: MockInstance<string, [foo: string], unknown>; }'.
    Types of property 'getHello' are incompatible.
      Type 'StubbedMember<(foo: string) => string>' is missing the following properties from type 'MockInstance<string, [foo: string], unknown>': getMockName, mock, mockClear, mockReset, and 13 more.ts(2322)
```
![MoreError](https://github.com/user-attachments/assets/499c0b32-84d0-422f-b87c-ca0708a0a720)

I checked the type files of `unitRef` and it wasn't modified. I also checked if the problem happens when installing diffent packages and it does.

Searching for a alternative, I chose to follow the recommendations from [NestJs Doc](https://docs.nestjs.com/recipes/automock), but I'd prefer to use the most recent version of `@suites` packages.

Any clue on what's going on?

**Comment by omermorad** [comment]
Hi @mbrunoviegas,
This occurs due to the `Mocked` type being imported from `@suites/doubles.jest` instead of being imported from `@suites/unit`.

This solution contains an issue; kindly refer to #426, which will be resolved shortly.

I would appreciate an update on how it turned out!

**Comment by qballer** [comment]
I would import Mocked like @omermorad suggested and use ```yarn install --force``` after adding a package as a workaround to the `postinstall` issue in yarn. You may refer to issue #426 for more details.

**Comment by omermorad** [comment, marked as answer]
@mbrunoviegas please refer here: https://github.com/suites-dev/suites/issues/426#issuecomment-2401114893
